FUNCTION-SPEC.md

```diff
@@ -0,0 +1,239 @@
+# Lambda-cfn function specification
+## Common to all function definitions
+- Only JavaScript based functions currently supported
+- All functions must export the function definition in the `function.template.js` file using the `lambda-cfn.build` function. The bare minimum function definition is:
+	```javascript
+	var lambdaCfn = require('lambda-cfn');
+
+	module.exports = lambdaCfn.build({
+		name: STRING_VALUE, /* required */
+	});
+
+	```
+- Lambda-cfn defaults to Nodejs v6.10.2, Nodejs 4.3.2 also available, see [AWS Lambda execution environment](http://docs.aws.amazon.com/lambda/latest/dg/current-supported-versions.html) for more information about the Lambda Nodejs runtime environment.
+- Lambda runtime parameters for `memorySize` and `timeout` are optional.
+    - `memorySize` must a multiple of 64MB between 128MB and 1536MB. If not specified, the default is 128mb. If the value specified is out of bounds, Lambda-cfn will round to the nearest valid memorySize value.
+    - `timeout` can be 0 to 300 seconds. If not specified, the default is 60 seconds. If the value specified is out of bounds, lambda-cfn will round to the nearest valid timeout value.
+
+	```javascript
+			var lambdaCfn = require('lambda-cfn');
+
+			module.exports = lambdaCfn.build({
+				name: STRING_VALUE, /* required */
+				runtime: 'nodejs4.3', /* optional, 'nodejs6.10' (default) or 'nodejs4.3' */
+				memorySize: '1536', /* in MB, optional, defaults to 128MB  */
+				timeout: '300' /* in seconds, optional, defaults to 60 seconds */
+			});
+
+		```
+- Lambda-cfn creates sane IAM policy defaults, but if additional policies are necessary for the function they should be specified in the `statements` array.
+	- example statement giving the function access to the AWS Support API:
+	```javascript
```

Comment:
@zmully - I previewed this code sample as regular Markdown and am seeing some strange indentation and spacing that makes this code sample hard to read. I'm seeing it in the code sample immediately below this one as well.

![screen shot 2017-08-14 at 6 10 44 pm](https://user-images.githubusercontent.com/10409657/29294167-05f8cea2-811c-11e7-86d4-a04bb9eff2e7.png)